code/modules/cargo/random_stock/t2_uncommon.dm

```diff
@@ -0,0 +1,378 @@
+// --- Uncommon ---
+
+STOCK_ITEM_UNCOMMON(glowshrooms, 2)
+	new /obj/item/seeds/glowshroom(L)
+	new /obj/item/seeds/glowshroom(L)
+	new /obj/item/seeds/glowshroom(L)
+
+STOCK_ITEM_UNCOMMON(plasteel, 3)
+	new /obj/item/stack/material/plasteel(L, rand(1,30))
+
+STOCK_ITEM_UNCOMMON(silver, 2)
+	new /obj/item/stack/material/silver(L, rand(5,30))
+
+STOCK_ITEM_UNCOMMON(phoronglass, 2)
+	new /obj/item/stack/material/glass/phoronglass(L, 50)
+
+STOCK_ITEM_UNCOMMON(sandstone, 2)
+	new /obj/item/stack/material/sandstone(L, 50)
+
+STOCK_ITEM_UNCOMMON(marble, 2)
+	new /obj/item/stack/material/marble(L, 50)
+
+STOCK_ITEM_UNCOMMON(iron, 2)
+	new /obj/item/stack/material/iron(L, 50)
+
+STOCK_ITEM_UNCOMMON(flare, 2)
+	new /obj/item/device/flashlight/flare(L)
+	new /obj/item/device/flashlight/flare(L)
+	if (prob(50))
+		new /obj/random/glowstick(L)
+
+STOCK_ITEM_UNCOMMON(deathalarm, 2)
+	new /obj/item/weapon/storage/box/cdeathalarm_kit(L)
+
+STOCK_ITEM_UNCOMMON(trackimp, 1)
+	new /obj/item/weapon/storage/box/trackimp(L)
+
+STOCK_ITEM_UNCOMMON(flashbang, 0.75)
+	new /obj/item/weapon/storage/box/flashbangs(L)
+
+STOCK_ITEM_UNCOMMON(cuffs, 1)
+	new /obj/item/weapon/storage/box/handcuffs(L)
+
+STOCK_ITEM_UNCOMMON(monkey, 2)
+	if (prob(40))
+		var/type = pick( \
+			/obj/item/weapon/storage/box/monkeycubes/farwacubes, \
+			/obj/item/weapon/storage/box/monkeycubes/stokcubes, \
+			/obj/item/weapon/storage/box/monkeycubes/neaeracubes \
+		)
+		new type(L)
+	else
+		new /obj/item/weapon/storage/box/monkeycubes(L)
+
+STOCK_ITEM_UNCOMMON(specialcrayon, 1.5)
+	if (prob(50))
+		new /obj/item/weapon/pen/crayon/mime(L)
+	else
+		new /obj/item/weapon/pen/crayon/rainbow(L)
+
+STOCK_ITEM_UNCOMMON(contraband, 2)
+	for (var/i in 1 to rand(1,8))
+		new /obj/random/contraband(L)
+
+STOCK_ITEM_UNCOMMON(mediumcell, 3)
+	for (var/i in 1 to rand(1,2))
+		var/type = pick( \
+			/obj/item/weapon/cell/super, \
+			/obj/item/weapon/cell/potato, \
+			/obj/item/weapon/cell/high \
+		)
+		new type(L)
+
+STOCK_ITEM_UNCOMMON(chempack, 5)
+	var/list/chems = SSchemistry.chemical_reagents.Copy()
+	var/list/exclusion = list("drink", "reagent", "adminordrazine", "beer2", "azoth", "elixir_life", "liquid_fire", "philosopher_stone", "undead_ichor")
```

Comment:
Missing a couple of chems that should be added to this list: "love", "shapesand", "usolve", "sglue", "black_matter", "lightning", "trioxin"